cmd/holo-files/internal/impl/entity_apply_nonorphan.go

```diff
@@ -36,23 +36,48 @@ import (
 //resources, and saving the result in the target path with the correct
 //file metadata.
 func (entity *Entity) applyNonOrphan(withForce bool) (skipReport bool, err error) {
-	//determine the related paths
-	var (
-		targetPath      = entity.PathIn(common.TargetDirectory())
-		basePath        = entity.PathIn(common.BaseDirectory())
-		provisionedPath = entity.PathIn(common.ProvisionedDirectory())
-	)
-
-	//step 1: entities may only be applied if:
-	//option 1: there is a manageable file in the target location (this target
-	//file is either the base from the application package or the
-	//product of a previous Apply run)
-	//option 2: the target file was deleted, but we have a base that we
-	//can start from
+	//step 1: check if a system update installed a new version of the stock
+	//configuration
+	//
+	// This has to come first because it might shuffle some files
+	// around, and if we do anything else first, we might end up
+	// stat()ing the wrong file.
+	newBasePath, newBase, err := entity.GetNewBase()
+	if err != nil {
+		return false, err
+	}
+
+	// step 2: Load our 3 versions into memory.
+	current, err := entity.GetCurrent()
+	if err != nil && !os.IsNotExist(err) {
+		if pe, ok := err.(*os.PathError); ok {
+			pe.Op, pe.Path = "skipping", "target"
+		}
```

Comment:
I would certainly say that `errors.New(...)` is better. I won't merge the current version because I stared at it for 5 minutes before understanding what's going on, and I'm sure I'll be doing the same when I touch that file again in 6 months. Re `github.com/pkg/errors`, I don't mind if you add it, but I don't see it adding any value over `errors.New()` either. In the end, what are we going to do with these errors anyway, if not stringifying them?